backend/app/api/endpoints/recommendation.py

```diff
@@ -0,0 +1,32 @@
+from fastapi import APIRouter, HTTPException
+
+import app.use_case.recommendation as recommendation
+from app.schema.recommendation import FreeFormatQuery, StructuredQuery
+
+router = APIRouter(
+    prefix="/recommendation",
+    tags=["generate_recommendation"],
+)
+
+
+@router.post("/structured-format")
+async def generate_recommendation(request_body: StructuredQuery) -> str:
+    try:
+        return recommendation.generate_recommendation_structured_format_query(
```

Comment:
I'm unsure if we want to return an error in a raw string form. Probably make it like

```
{
status: "success", // or failed
message: "xyz",
error: "Give actual logs of the error here or a desired error"
}
```
If we want a structured error, we can make a constant enum for error.
The error can later be displayed via snackbar in the frontend.